khelo/lib/ui/flow/home/home_view_model.dart

```diff
@@ -46,6 +54,27 @@ class HomeViewNotifier extends StateNotifier<HomeViewState> {
     );
   }
 
+  Map<MatchStatusLabel, List<MatchModel>> _groupMatches(
+      List<MatchModel> matches) {
+    final groupedMatches = groupBy(matches, (match) {
+      switch (match.match_status) {
+        case MatchStatus.running:
+          return MatchStatusLabel.live;
+        case MatchStatus.yetToStart:
+          return MatchStatusLabel.upcoming;
+        case MatchStatus.abandoned:
+        case MatchStatus.finish:
+          return MatchStatusLabel.winning;
+      }
+    });
+    return {
+      MatchStatusLabel.live: groupedMatches[MatchStatusLabel.live] ?? [],
+      MatchStatusLabel.upcoming:
+          groupedMatches[MatchStatusLabel.upcoming] ?? [],
+      MatchStatusLabel.winning: groupedMatches[MatchStatusLabel.winning] ?? [],
+    };
```

Comment:
It's not the same because we need to display the map according to live, upcoming, and finished statuses. If we return the group directly, we won't get the correct order, which is why we need to specify the positions manually.